substrate/frame/recovery/src/lib.rs

```diff
@@ -705,6 +702,43 @@ pub mod pallet {
 			Ok(())
 		}
 	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		#[cfg(feature = "try-runtime")]
+		fn try_state(_n: BlockNumberFor<T>) -> Result<(), sp_runtime::TryRuntimeError> {
+			Self::do_try_state()
+		}
+	}
+}
+
+
+#[cfg(any(feature = "try-runtime", test))]
+impl<T: Config> Pallet<T> {
+	/// Ensure the correctness of the state of this pallet.
+	///
+	/// This should be valid before or after each state transition of this pallet.
+	pub fn do_try_state() -> Result<(), sp_runtime::TryRuntimeError> {
+		Self::try_state_friends()?;
+
+		Ok(())
+	}
+
+	/// # Invariants
+	///
+	/// `Friends` should not exceed the `MaxFriends` capacity.
+	fn try_state_friends() -> Result<(), sp_runtime::TryRuntimeError> {
+		Recoverable::<T>::iter().try_for_each(|(_, recovery_config)| -> DispatchResult {
+			ensure!(
+				recovery_config.friends.len() as u32 <= T::MaxFriends::get(),
```

Comment:
`friends` is a bounded vec, which are guaranteed on a type level to never exceed their allowed length:  

https://github.com/paritytech/polkadot-sdk/blob/5a18f8b154b9dfa31ece22096277ba03dac58094/substrate/frame/recovery/src/lib.rs#L181

So i dont see how this is useful.

Reply:
@ggwpez Any suggestion?